Backendless/WebORB/Client/HttpEngineWithClient.cs

```diff
@@ -0,0 +1,220 @@
+using System.Net;
+using System.Net.Http.Headers;
+using Weborb.Types;
+using Weborb.Protocols.Amf;
+using Weborb.Exceptions;
+
+namespace Weborb.Client
+{
+  public class HttpEngineWithClient : Engine
+  {
+    HttpClient httpClient = new HttpClient();
+    HttpWebRequest _request;
+    public HttpEngineWithClient( String url, IdInfo idInfo ) : base( url, idInfo )
+    {
+    }
+
+    public override void SendRequest<T>( V3Message v3Msg, IDictionary requestHeaders,
+                                         IDictionary httpHeaders, Responder<T> responder,
+                                         AsyncStreamSetInfo<T> asyncStreamSetInfo ) =>
+      ThreadPool.QueueUserWorkItem( state => SendHttpRequest( v3Msg, requestHeaders, httpHeaders, responder, asyncStreamSetInfo ) );
+
+    internal override void Invoke<T>( string className, string methodName, object[] args, IDictionary requestHeaders,
+                                  IDictionary messageHeaders, IDictionary httpHeaders, Responder<T> responder,
+                                  AsyncStreamSetInfo<T> asyncStreamSetInfo ) =>
+      SendRequest( CreateMessageForInvocation( className, methodName, args, messageHeaders ),
+                                requestHeaders, httpHeaders, responder, asyncStreamSetInfo );
+
+    public override Task<T> SendRequest<T>( V3Message v3Msg, IDictionary requestHeaders, IDictionary httpHeaders,
+                                                                  ResponseThreadConfigurator threadConfigurator ) =>
+      SendHttpRequest<T>( v3Msg, requestHeaders, httpHeaders, threadConfigurator );
+
+    internal override Task<T> Invoke<T>( string className, string methodName, object[] args, IDictionary requestHeaders,
+               IDictionary messageHeaders, IDictionary httpHeaders, ResponseThreadConfigurator threadConfigurator ) =>
+      SendRequest<T>( CreateMessageForInvocation( className, methodName, args, messageHeaders ),
+                                        requestHeaders, httpHeaders, threadConfigurator );
+
+    private async void SendHttpRequest<T>( V3Message v3Msg, IDictionary requestHeaders, IDictionary httpHeaders, Responder<T> responder, AsyncStreamSetInfo<T> asyncStreamSetInfo )
+    {
+      byte[] requestBytes = CreateRequest( v3Msg, requestHeaders );
+      HttpRequestMessage requestMessage = new HttpRequestMessage
+      {
+        RequestUri = new Uri( GatewayUrl ),
+        Method = HttpMethod.Post,
+        Content = new ByteArrayContent( requestBytes )
+      };
+
+      requestMessage.Content.Headers.ContentType = new MediaTypeHeaderValue( "application/x-amf" );
+      HttpResponseMessage responseMessage = await httpClient.SendAsync( requestMessage );
+
+      asyncStreamSetInfo.responder = responder;
+
+      var streamResponse = await responseMessage.Content.ReadAsStreamAsync();
+      var curTime = DateTime.Now.Ticks;
+      var roundTrip = ( curTime - asyncStreamSetInfo.messageSentTime ) / TimeSpan.TicksPerMillisecond;
```

Comment:
where is this used?

Reply:
If the operation ends with an error, then using this we catch it

Reply:
the question is for line 62

Reply:
The fields in which the operation execution time is stored, in general they are not used anywhere, they can be cut